solr/core/src/test/org/apache/solr/response/TestRawTransformer.java

```diff
@@ -16,55 +16,151 @@
  */
 package org.apache.solr.response;
 
+import org.apache.commons.io.FileUtils;
 import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.client.solrj.embedded.JettySolrRunner;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.impl.NoOpResponseParser;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.request.QueryRequest;
+import org.apache.solr.cloud.MiniSolrCloudCluster;
+import org.apache.solr.cloud.SolrCloudTestCase;
 import org.apache.solr.common.SolrInputDocument;
-import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.common.params.ModifiableSolrParams;
 import org.junit.After;
+import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
+import java.io.File;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Properties;
+
 /**
  * Tests Raw JSON output for fields when used with and without the unique key field.
  *
  * See SOLR-7993
  */
-public class TestRawTransformer extends SolrTestCaseJ4 {
+public class TestRawTransformer extends SolrCloudTestCase {
+
+  private static final String DEBUG_LABEL = MethodHandles.lookup().lookupClass().getName();
+
+  /** A basic client for operations at the cloud level, default collection will be set */
+  private static JettySolrRunner JSR;
+  private static HttpSolrClient CLIENT;
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    initCore("solrconfig-doctransformers.xml", "schema.xml");
+    if (random().nextBoolean()) {
```

Comment:
ONe more thing, is this a common pattenr for doing solrcloud versus NON solr cloud?   It just looks a bit odd to me, is there a better way of swapping the two?

Reply:
Heh; there isn't really a "common pattern" for doing solrcloud vs standalone -- if you can find a better example to pattern after I'm certainly open to alternatives. iiuc this is what [SOLR-11872](https://issues.apache.org/jira/browse/SOLR-11872) sets out to do. This would have been _far_ easier also if it weren't for the fact that we specifically need to exercise the non-javabin serialization formats, and inspect their output directly. So that's _part_ of what made this setup more complex.